pallets/nft/src/benchmarking.rs

```diff
@@ -26,164 +28,80 @@ fn dollar(d: u32) -> u128 {
 	d.saturating_mul(100_000_000_000_000)
 }
 
+fn do_create_class<T: Config>(
+	class_id: T::NftClassId,
+) -> (T::AccountId, <T::Lookup as StaticLookup>::Source, BoundedVecOfUnq<T>) {
+	let caller = create_account::<T>("caller", 0);
+	let caller_lookup = T::Lookup::unlookup(caller.clone());
+	let metadata: BoundedVec<_, _> = vec![0; <T as UNQ::Config>::StringLimit::get() as usize]
+		.try_into()
+		.unwrap();
+	assert!(NFT::Pallet::<T>::create_class(
+		RawOrigin::Signed(caller.clone()).into(),
+		class_id,
+		Default::default(),
+		metadata.clone()
+	)
+	.is_ok());
+	(caller, caller_lookup, metadata)
+}
+
+fn do_mint<T: Config>(class_id: T::NftClassId, instance_id: T::NftInstanceId) {
+	let caller = create_account::<T>("caller", 0);
+	let metadata: BoundedVec<_, _> = vec![0; <T as UNQ::Config>::StringLimit::get() as usize]
+		.try_into()
+		.unwrap();
+	assert!(NFT::Pallet::<T>::mint(RawOrigin::Signed(caller).into(), class_id, instance_id, metadata).is_ok());
+}
+
 benchmarks! {
 	create_class {
 		let caller = create_account::<T>("caller", 0);
-		let big_vec = vec![1; <T as orml_nft::Config>::MaxClassMetadata::get() as usize];
-		let class_metadata = big_vec;
-		let class_data = ClassData { is_pool:false };
-		let class_id = orml_nft::Pallet::<T>::next_class_id();
-	}: _(RawOrigin::Signed(caller.clone()), class_metadata, class_data)
-	verify {
-		assert_eq!(orml_nft::Pallet::<T>::classes(class_id).iter().count(), 1);
-	}
-
-	create_pool {
-		let caller = create_account::<T>("caller", 0);
-		let big_vec = vec![1; <T as orml_nft::Config>::MaxClassMetadata::get() as usize];
-		let class_metadata = big_vec;
-		let class_data = ClassData { is_pool:true };
-		let class_id = orml_nft::Pallet::<T>::next_class_id();
-		let price: BalanceOf<T> = u32::MAX.into();
-	}: _(RawOrigin::Signed(caller.clone()), class_metadata, class_data, price)
+		let metadata: BoundedVec<_, _> = vec![0; <T as UNQ::Config>::StringLimit::get() as usize].try_into().unwrap();
+	}: _(RawOrigin::Signed(caller.clone()), CLASS_ID_0.into(), Default::default(), metadata)
 	verify {
-		assert_eq!(orml_nft::Pallet::<T>::classes(class_id).iter().count(), 1);
+		assert_eq!(UNQ::Pallet::<T>::class_owner(&T::NftClassId::from(CLASS_ID_0).into()), Some(caller));
```

Comment:
Is this the worst case? Is adding to full vec more expensive? > https://substrate.stackexchange.com/questions/88/is-there-a-limit-to-the-size-of-a-map-in-pallet-storage

Reply:
It is adding to a map, not to a vec. Testing adding stuff to a nearly full map might be an overkill. There should be another incentives to keep the map size reasonable, in this case there is a configurable bond to create a class that exists until the class is destroyed.